snippets/graph/mst.cpp

```diff
@@ -0,0 +1,39 @@
+struct edge{
+	int u,v,w;
+	edge (int _, int __, int ___) { u = _, v = __, w = ___;}
+};
+
+// returns ptr to vector of edges in the mst, reasonably fast
+// graph[i] = vector of edges incident to vertex i
+// places total weight of the mst in &total
+vector<edge> *mst(vector<edge> *graph, size_t n, size_t m, long long &total) {
```

Comment:
If we're using vectors anyway I prefer to fully use it and go `vector<vector<edge>> graph` instead of `vector<edge> *graph, size_t n` but that's just me